### ISSUE TITLE
Problem with SQl Where IN clause and subselect

### ISSUE THREAD

**Issue by EricSchreiner** [orientechnologies/orientdb#8745]
## Expected behavior  
See attached Screenshot from Orientdb studio

## Actual behavior  
See attached Screenshot from Orientdb studio

## Steps to reproduce  
See attached Screenshot from Orientdb studio

![select](https://user-images.githubusercontent.com/4680347/51913423-b9e0c600-23d6-11e9-8868-88f4185a23bb.jpg)

**Comment by luigidellaquila** [comment]
Hi @EricSchreiner 

The first query doesn't work because you are matching a `key` with a pair `key, rid`.
If you change it as follows it will work fine

```
SELECT key, rid from index... where key in (select key from index...)
```

Thanks

Luigi

**Comment by EricSchreiner** [comment]
Hi @luigidellaquila 
(I would reopen but it seems I'm not allowed to.) I've tried a lot of different solutions but it seems to be a bug. See Screenshot below with your suggestion....
![select-2](https://user-images.githubusercontent.com/4680347/52126822-af216d80-2630-11e9-80fd-bb44f6ab74ee.jpg)

**Comment by luigidellaquila** [comment]
Hi @EricSchreiner 

I tried it with DemoDB on v 3.0.14 and it seems to work fine, maybe it's already fixed. Could you please try your dataset on latest stable?

Thanks

Luigi

**Comment by EricSchreiner** [comment]
Hi @luigidellaquila my example was already running on 3.0.14

**Comment by EricSchreiner** [comment]
Hi @luigidellaquila 
for you information: It works fine when i use a static list of keys in the IN clause instead of a select....

**Comment by luigidellaquila** [comment]
Hi @EricSchreiner 

Can you please give me more details on how to reproduce?
I tried the following on DemoDB and the result is as expected

```
select from index:Countries.Id where key in (select key from index:Countries.Id where key = 1)
```

What is different in your case?

Thanks

Luigi

**Comment by EricSchreiner** [comment]
Hi @luigidellaquila ,
my key is OType.STRING and duplicates are allowed (INDEX_TYPE.NOTUNIQUE)

**Comment by luigidellaquila** [comment]
Hi @EricSchreiner 

Thank you very much, I managed to reproduce the problem. I'll try to fix it ASAP

I'll keep you updated

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @EricSchreiner 

I just pushed a fix, it will be in the snapshot in a few minutes and will be released with v 3.0.15

Thanks

Luigi

**Comment by EricSchreiner** [comment]
Hi @luigidellaquila 
thanks....
